### ISSUE TITLE
TCL4 branch:  UVRs should be stored in grid

### ISSUE THREAD

**Issue by nasajoey** [wing-aviation/InterUSS-Platform#39]
**Problem**
To support future operations, there is a gap related to knowing about any dynamic airspace restrictions (to be known as UAS Volume Reservations or UVRs).

**Proposed Solution**
_This is a TCL4 problem/solution._  We should explore storing UVRs directly in the grid.  This would allow a new entrant to a given grid to immediately have access to the state of that airspace.  UVRs should only be writeable by USSs with an enhanced role, thus we'd have to check for a different scope in the access_token provided.  Only the USS that writes the UVR would have rights to modify or delete.  We may want a feature that automatically expires/deletes UVRs after some time (like within an hour of the UVR's end time), but we may want to wait on that and track as an additional github issue.

**Describe alternatives you've considered**
Alternatively, one could force enhanced USSs to serve these data to other USSs upon request.  There are issues with this approach.  First and foremost, how do we guarantee that such a USS is always available in a given grid?  And how are they easily identified from other USSs in the grid (some new metadata tag?)?

**Additional context**
Will need this sooner than later.  We can maybe target Sprint 4, but would take it whenever it might be available.

The model written to grid may leverage the existing UVR schema:  https://github.com/nasa/utm-apis/blob/3189acdb3ce13dd64258cecb0e4f3c40c158a202/utm-domains/utm-domain-commons.yaml#L1270

**Comment by myspri** [comment]
@BenjaminPelletier, we have updated UVR model 
https://github.com/nasa/utm-apis/blob/v4-draft/utm-domains/utm-domain-commons.yaml

This is usually announced in the uss_implementor's channel.
Going forward, we will give you advance notice that will allow our models to be in sync before the announcement goes out.

Can you please update your model based on this?

We have been trying to codegen from your version and have encountered a problem with "allOf" attribute. We are currently blocked. I believe the v4-draft does not include this attribute.

Hopefully, this problem will go away after you sync up with our version.

**Comment by drsmit22** [comment]
I have also noticed a "timestamp" field being added to UASVolumeReservation when it is being returned from the grid that is causing trouble.

**Comment by issmith1** [comment]
If the grid has old prior-version models that are returned via GET, that can happen.   We need to clean out incompat models.

**Comment by BenjaminPelletier** [comment]
Please open a new issue for each new issue.  This issue has already been resolved.

**Comment by BenjaminPelletier** [comment]
@issmith1, while the first round of UVR changes (storage API version 4.0.3 -> 4.1.0) are API-incompatible, they should be grid-compatible.  In other words, grid content should not need to be cleared or updated to use the 4.1.0 or 4.1.1.
